### ISSUE TITLE
Should message returned from `check_factor()` refer to `.f` instead of `f`?

### ISSUE THREAD

**Issue by batpigandme** [tidyverse/forcats#260]
Branching this off of #259, the current error message returned by `check_factor()` references `f`:
```r
<error/rlang_error>
`f` must be a factor (or character vector).
Backtrace:
  9. forcats::fct_reorder(., method.source, value2, median)
 10. forcats:::check_factor(.f)
```
The first argument of `fct_reorder()` is `.f`, as it is for `check_factor()`, which makes me think the error message should be:
```
`.f` must be a factor (or character vector).
```

Although now I'm a little confused since, looking at the source code, it looks like the first argument of `check_factor()` is `f` (which isn't what it looks like in the backtrace:
https://github.com/tidyverse/forcats/blob/5dbf80107bec1ab23d9e929923b78d85cc9fa380/R/utils.R#L14-L22

Happy to do a PR if it _does_ need to be changed!

**Comment by hadley** [comment]
`check_factor()` needs another argument that provides the argument name used in the top-level function, and then any call of the form `check_factor(.f)` needs to be changed to `check_factor(.f, ".f")`

**Comment by hadley** [comment]
I accidentally fixed this in 00bef41c6ef04f5b52be34ad8be86f1a8a220242
